cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -244,28 +255,53 @@ protected Future<Void> certificatesSecret(Clock clock) {
+                            Util.decodeFromBase64(topicOperatorApiSecret.getData().get(API_TO_ADMIN_PASSWORD_KEY)))
+                            : null;
+                        // generate a new CC API secret if there is no CC auth file, or there is no TO API secret, or TO admin password changed
+                        Secret newSecret = cruiseControlAuthFile == null || toAdminUser == null || !cruiseControlAuthFile.contains(toAdminUser.password())
+                            ? cruiseControl.generateApiSecret(passwordGenerator, null, toAdminUser)
+                            : cruiseControl.generateApiSecret(passwordGenerator, oldSecret, null);
+                        this.apiSecretHash = ReconcilerUtils.hashSecretContent(newSecret);
+                        return secretOperator.reconcile(reconciliation, reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()), newSecret)
+                            .map((Void) null);
+                    }
+                );
+            } else {
+                return secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()))
                     .compose(oldSecret -> {
-                        Secret newSecret = cruiseControl.generateApiSecret(passwordGenerator);
-
-                        if (oldSecret != null)  {
-                            // The credentials should not change with every release
-                            // So if the secret with credentials already exists, we re-use the values
-                            // But we use the new secret to update labels etc. if needed
-                            newSecret.setData(oldSecret.getData());
-                        }
-                        
+                        Secret newSecret = cruiseControl.generateApiSecret(passwordGenerator, oldSecret, null);
                         this.apiSecretHash = ReconcilerUtils.hashSecretContent(newSecret);
-                        
                         return secretOperator.reconcile(reconciliation, reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()), newSecret)
-                                .map((Void) null);
+                            .map((Void) null);
                     });
+            }
         } else {
             return secretOperator.reconcile(reconciliation, reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()), null)
                     .map((Void) null);
         }
     }
 
+    private static Future<Secret> getSecret(SecretOperator secretOperator, String namespace, String secretName)  {
+        return secretOperator.getAsync(namespace, secretName).compose(secret -> {
+            if (secret != null) {
+                return Future.succeededFuture(secret);
+            } else {
+                return Future.succeededFuture();
+            }
+        });
+    }
```

Comment:
What is this good for? `getAsync()` IIRC returns `null` when the secret does not exist. So:
* Why is this method needed?
* Why does it need the if-else instead of just returning?

Reply:
Right, included getAsync logic directly in apiSecret.